src/test/java/dk/sdu/se_f22/sortingmodule/range/rangepublic/DoubleFilterTest.java

```diff
@@ -63,37 +106,322 @@ void useFilter() {
+                        "Test Name",
+                        "Test Discription",
+                        "Test product Attribute",
+                        1, 1000);
+
+                Assertions.assertThrows(
+                        IllegalMinMaxException.class,
+                        () -> doubleFilter.setUserMax(input));
+            }
+
+            @ParameterizedTest
+            @MethodSource("provideParameters")
+            @DisplayName("Set invalid DoubleFilter userMin and then userMax")
+            void setValidDoubleFilterUserMinAndThenUserMax(double inputMin, double inputMax) throws IllegalMinMaxException {
+                DoubleFilter doubleFilter = new DoubleFilter(
+                        "Test Name",
+                        "Test Discription",
+                        "Test product Attribute",
+                        1, 1000);
+
+                doubleFilter.setUserMin(inputMin);
+
+                Assertions.assertThrows(
+                        IllegalMinMaxException.class,
+                        () -> doubleFilter.setUserMax(inputMax));
+            }
+
+            @ParameterizedTest
+            @MethodSource("provideParameters")
+            @DisplayName("Set invalid DoubleFilter userMax and then userMin")
+            void setValidDoubleFilterUserMaxAndThenUserMin(double inputMin, double inputMax) throws IllegalMinMaxException {
+                DoubleFilter doubleFilter = new DoubleFilter(
+                        "Test Name",
+                        "Test Discription",
+                        "Test product Attribute",
+                        1, 1000);
+
+                doubleFilter.setUserMax(inputMax);
+
+                Assertions.assertThrows(
+                        IllegalMinMaxException.class,
+                        () -> doubleFilter.setUserMin(inputMin));
+            }
+
+            private static Stream<Arguments> provideParameters() {
+                return Stream.of(
+                        Arguments.of(2.0, 1.0),
+                        Arguments.of(5.5, 3.5),
+                        Arguments.of(1000.0, 999.0)
+                );
+            }
+        }
+
+        @Nested
+        @DisplayName("Testing set userMin and userMax with incorrect types")
+        class testUserValuesWithIncorrectTypes {
+
+            static Stream<DoubleFilter> doubleFilterProvider() {
+                return provideDoubleFiltersProvider();
+            }
+
+            @ParameterizedTest
+            @DisplayName("Set userMin with incorrect type")
+            @MethodSource("doubleFilterProvider")
+            void setUserMinWithIncorrectType(DoubleFilter filter) {
+                long newValue = 100;
+
+                Assertions.assertThrows(IllegalMinMaxException.class,
+                        () -> filter.setUserMin(newValue)
+                );
+            }
+
+            @ParameterizedTest
+            @DisplayName("Set userMax with incorrect type")
```

Comment:
I would love for these tests, to actually have a name that describes what is tested rather than how it is tested.
For this specific test it may be along the lines of:
"setting incorrect type of userMax throws IllegalMinMaxException"
Or something like that